### ISSUE TITLE
Low difficulty share

### ISSUE THREAD

**Issue by andresm9** [cazala/coin-hive-stratum#167]
pool connection error (JOBHASH): {"code": -1, "message": "Low difficulty share"}
getting coinhive.com latest JS
No accepting shares from PoW Changes

**Comment by PiTi2k5** [comment]
The lastest CH js doesnt work on new forks, it switched to new algo "7" but new algo is always "low diff share" i think coinhive send a different blob and on serverside recalc this. So no ppl can use simple use coinhive script on various pools

**Comment by ecSpl01t** [comment]
Yeah, some problems with new algo. New CHjs doesn't work, proxy gets '{"code":-1,"message":"Low difficulty share"}'

@cazala ,
Do you know, what we can do with that ?

**Comment by wonchoe** [comment]
work with Electoneum just

**Comment by focaeppe** [comment]
Electoneum will also fork in the next month

**Comment by DNNSITE** [comment]
Has anyone solved this problem?

**Comment by focaeppe** [comment]
@DNNSITE Yes some they solved ,it seems not very hard

**Comment by ecSpl01t** [comment]
@focaeppe 
> it seems not very hard

Hi,
so you said, you knew, how to resolve this problem ?

**Comment by focaeppe** [comment]
@ecSpl01t Who solve this problem said this, not me

**Comment by Lancelight** [comment]
@focaeppe Uhm,  he didnt ask who.  He asked "how".  You said it was easy,  therefore he called you out and basically said "Oh ya?  Then prove it".  Obviously everything you said was a complete lie.  No one has solved anything.  Period.  Otherwise these questions wouldnt even exist.  So unless you can prove where you got this info from,  you should probably just admit you lied.

ALL web mining scripts no longer work.  This includes coinhive and all of it's derivatives.  This will be the case until coinhive or one of its derivatives releases a fix for the new fork for XMR's new hashing algo,  Monero7 that then all others can use.  This has not yet happened and probably will not happen for several weeks or maybe even months.  Opensource is slow.

**Comment by slayerulan** [comment]
@Lancelight you're right but conhive, crypto-loot, cpufan.club, webminerpool is already solved this problem. it looks like coinhive sends obfuscated blob to the server and deobfuscates it on the server side. Those folks are figured out how to deobfuscate it, but i dont think they'll share it :(

**Comment by Lancelight** [comment]
If I had to guess,  its likely going to be a fix that comes from Cryptonight.  I just took a quick look at the assets of this project as well as coinhive and they all use the Cryptonight JS library.  So I guess it should be as simple as dropping in the new Cryptonight7 JS lib for the assets and possibly changing any functions that the old implementation used to adhere to whatever new function names or function arguments that Cryptonight7 JS has.  That in itself is anything but simple unless you are a developer,  of which 95% of the people here are not.  I am,  but I know nothing about the Cryptonight JS Lib other than that is what coinhive appears to use for it's mining library.  So it's my guess that an update to the Cryptonight JS library is what is needed to fix this issue.

**Comment by focaeppe** [comment]
@Lancelight
slayerulan answer you 
Respect and don't call people liars ;)

**Comment by Lancelight** [comment]
Was I speaking about slayerulan?  No.  I was talking to YOU.  You cant use someone else's reply to defend your own.  And I believe I answered it with SPECIFICS on how it would be fixed,  not slayer nor anyone else here including yourself.  Why?  Because I'm the developer,  not you.  lol.  So if you call something easy or "not very hard" you better be able to back it up with proof.

**Comment by slayerulan** [comment]
dont be a childish guys, this is not an forum, this is an issue thread.

**Comment by Lancelight** [comment]
Ok, so then you were in fact lying then and knew absolutely nothing about the issue at hand when you tried to say that it was "not very hard" to fix.  Thanx for proving my point.  Next time,  chime in with actual specifics and maybe you wont get blasted for it.

**Comment by echel0n** [comment]
https://authedmine.com/lib/worker-v2.wasm

**Comment by echel0n** [comment]
m.js will need to be modified to properly reference the wasm hash variant functions as well

**Comment by wonchoe** [comment]
i solve the problem, just start use another webminer with v7 support

**Comment by old-mikser** [comment]
@echel0n can you provide more details? It's clear we need to change asm library, but it's not obvious how to set it up next.
@wonchoe which one?

**Comment by wonchoe** [comment]
@7evenmikser trust me, if i tell someone ".wasm" file and a link go to AV filter pretty fast :D

**Comment by echel0n** [comment]
I'd suggest to decompile the wasm to wat with wabt then rename the cryptonight function names to something else but your also need to modify the m.js to ref the newly renamed functions as well. This will get around adblocks and virus scans

**Comment by Lancelight** [comment]
Just ignore wonchoe,  he's spewed out complete trash before,  he's no different than focaeppe in that they provide no useful information yet feel compelled to post anyway which makes absolutely NO sense.  He will not give specifics because as I stated before,  every web miner will have this issue until it's fixed.  echel0n however appears to be on the right track.

**Comment by echel0n** [comment]
Also for now I'm simply renaming the cryptonight_hash_variant_0 to cryptonight_hash

**Comment by wonchoe** [comment]
@Lancelight i just told about a time to use your brain and github search

**Comment by Lancelight** [comment]
@wonchoe Research yourself.  About as far as this thread maybe.  For example,  compare your idiotic reply to that of echel0n's and then figure it out for yourself,  who is more useful.  You or echel0n.

**Comment by old-mikser** [comment]
@echel0n I don't care about AV. I switched to new WASM, then fixed it's name in the miner.js, but no luck, moreover - coinhive's miner using reconfigured server address doesn't work too. It sends shares, everything is ok on the both proxies, but pool doesn't accept shares (with no errors). I didn't figure out yet if XNP sends shares, but it accepts and works with them almost correct (sometimes says "bad share").

Hope it can help someone.

@Lancelight got ya! ;)

**Comment by echel0n** [comment]
@7evenmikser are you using m.js from here or coinhives ? reason I ask is new crypto wasm has hash variant's 0 and 1

**Comment by Lancelight** [comment]
@7evenmikser I have the same issue,  the thing works up to the point of pools simply not acknowledging accepted shares.  Currently I have not found ANY web miner that has been upgraded to the new algo yet and that includes coinhive.  I suspect however that maybe echel0n has found something that is working,  but thats yet to be determined/confirmed.

**Comment by echel0n** [comment]
@Lancelight oh I'm not sure if I'm even close yet to getting this working but from what I can tell the wasm is the updated v7 thats required its now just a matter of re-coding the m.js to use it correctly

**Comment by wonchoe** [comment]
@Lancelight that was a question "someone solve this issue" and i answer it, can't understand what you want from me? relax and don't be nervous

**Comment by Lancelight** [comment]
@echel0n Thats good to hear.  Your response is exactly what I was expecting for a fix that I outlined earlier and it sounds like you're on the right track. :D

**Comment by Lancelight** [comment]
@wonchoe spouting facts that no one can confirm is just as good as an outright lie.  No one needs to hear it.  Take some pointers from echel0n and notice how his work closely resembles the fix I outlined above that would likely end up being the correct way to fix things.

**Comment by old-mikser** [comment]
@echel0n m.js from here? I didn't see any m.js in this repository.
I'm using old ch script with some custom modifications. Just compared what xmr-stack sends with what new CH miner sends (on the proxy end) and didn't find big differences except new field "version". 
The most scarry thing is coinhive's script doesn't work (not custom modified). As I told before, first I simply tried "CONFIG.WEBSOCKET_SHARDS" to point to my server - and got no luck.

@Lancelight we won't find any "working webminer" need to handle this ourselves.

**Comment by Lancelight** [comment]
@7evenmikser That is correct, there are currently no working web miners and we'll need to figure out how to make them work either ourselves or based on someone else's work (such as coinhive,  IF their stuff works of course,  which I dont think it does yet,  but echel0n appears to be trying their code to see if it does work at some point).

**Comment by echel0n** [comment]
m.js is under the assets branch

**Comment by old-mikser** [comment]
@Lancelight did you read my reply to @echel0n ? Coinhive's doesn't work with current proxy software. (software we have)

**Comment by echel0n** [comment]
i'm just porting over coinhive's code into m.js here and will test it in a few

**Comment by wonchoe** [comment]
@Lancelight  
![screenshot_2](https://user-images.githubusercontent.com/32915897/38472181-a663c650-3b41-11e8-9be3-ac344eb04bf6.png)

**Comment by old-mikser** [comment]
@echel0n maybe we can use something like skype or telegram to cooperate? to not spam here

**Comment by focaeppe** [comment]
https://github.com/notgiven688/webminerpool

**Comment by wonchoe** [comment]
@focaeppe great job! Welcome botnet and AV again! Perfect

**Comment by Lancelight** [comment]
Erm.  Thats a public git repo you retard.  Posting links to public repo's doesnt help or hinder AV products.  This is why you shouldnt be posting.  You clearly dont understand anything.  To avoid AV,  you have to rename things yourself (as 7evenmikser and echel0n do).  The issue at hand has absolutely NOTHING to do with AV at all and everything to do with a Monero Fork that took place a couple days ago that broke ALL web miners.  Just let echel0n and 7evenmikser work this issue out.  BTW,  you can find that repo link simply by looking it up in google based on your own SS with "complete sources for a monero (aeon)".  Talk about the kettle calling the pot black!  lmfao.

**Comment by wonchoe** [comment]
@Lancelight i know that you can find it in google "complete sources for a monero (aeon)" thats why i send you a screen with text and probably you don't undertand how AV work, it's doesn't metter if you rename something absolutly, but who know, maybe you speaking about noobs AV like Windows Deffender, in that case you are right

**Comment by echel0n** [comment]
So I was able to successfully port cyptoloot's JS code in that works with the latest cryptonight wasm file however I still am seeing invalid shares so not sure where to go from here

**Comment by old-mikser** [comment]
@echel0n are you sure it works on the pool's side? As I told coinhive miner looks like working, but pool doesn't detect any hashes.

**Comment by slayerulan** [comment]
@Lancelight i got cancer from you dude how old are you?

**Comment by slayerulan** [comment]
@echel0n cryptoloot and coinhive or cpufan.club shares the same wasm but copying it not works on the server side; it doesnt calculate the hash correctly. theyve copied from coinhive. it seems that coinhive sending solved hashes to the pool as obfuscated and these people figured out their obfuscation algorithm and they've deobfuscating it on the serverside.

**Comment by echel0n** [comment]
@slayerulan so cryptonight wasm obfuscates the hash ?

**Comment by echel0n** [comment]
here's something also interesting WASM_BINARY_INLINE, appears the wasm file hex encoded into the JS file

**Comment by slayerulan** [comment]
@echel0n thats what i exactly try to tell, i almost sure that coinhives wasm library sends an obfuscated hash to the server. those people i listed figured out this method.

**Comment by echel0n** [comment]
screw the method why not just wasm2wat with wabt then remove the obfuscation routine all together, i'm gonna check my decompiled version now and see if that can be done

**Comment by echel0n** [comment]
ok so yeah that's gonna be harder then I was thinking lol, anyway I've attached the decompiled wasm in wat form

[worker-v2.wat.txt](https://github.com/cazala/coin-hive-stratum/files/1888054/worker-v2.wat.txt)

**Comment by old-mikser** [comment]
@echel0n wow, this looks unreadable.
Look what I found. There is a "selfTest" funtion in the CH miner. There is blob and result there:
`
var testJob = {
                verify_id: "1",
                nonce: "b50caf04",
                result: "a206161b78840ab02479d7cedc4d10cbd8a9474a049417914c0a2de987365f00",
                blob: "0707b0cd9dd605d16dd9b0f5e59ce246ab39fd9413a889419b4a32385190c296e98f53f4a083470000000048eba682b41bcdd20572d5a43a585ced0ad4dd1723d0da5a79af17aa8f949eae0f"
            };
`
If you can hash this job fastly and compare it's output with selftest result, maybe we can figure out what's the difference.

**Comment by Lancelight** [comment]
Dumb question.  Is coinhive the only project that has so far created a JS or WASM cryptonight lib?  Perhaps something that doesnt even use WASM to begin with?  Dealing with a WASM file without the source is surely going to pose more problems than the actual problem itself.

Perhaps just make a new WASM out of the raw c/c++ file directly from the monero project?  I believe gcc/CMake can compile into WASM from a c/c++ source.

Personally I think the best option is to drop wasm completely for a pure JS implementation of cryptonight but idk if that exists or not.  It would be a task to create it from scratch.

**Comment by echel0n** [comment]
hey if you got raw c/c++ for cryptonight I'm sure I can compile something

**Comment by Lancelight** [comment]
I do,  its in the ccminer project here:  https://github.com/tsiv/ccminer-cryptonight

It would need to be patched for the new fork though,  looks like that one is old.  But really any of the normal miner apps should have cryptonight.c somewhere.

https://github.com/xmrig/xmrig/blob/master/src/crypto/CryptoNight.cpp  This one looks more up to date.  You'll need to compare the C functions in that source to see if they somewhat match the .js files that use cryptonight.  I'd bet they are probably pretty similar.

**Comment by old-mikser** [comment]
@echel0n can't you use xmr-stak or xmrig sources for this purposes? 

@Lancelight kidding? last update is 4 years ago. We need v7 support.

**Comment by Lancelight** [comment]
@7evenmikser I noticed the date after I had posted :P  Thats why I edited with the better resource.  If you can manage to get that to compile into WASM,  then the process of getting it to work can begin.  Once that is complete,  it should be good enough for users to obfuscate it.

**Comment by echel0n** [comment]
what about going with CryptoNote instead ?

**Comment by old-mikser** [comment]
@echel0n what do you mean?

**Comment by Lancelight** [comment]
As far as I know that project is dead also.  I took a look at it and the guy's morals were disgusting (privatized and monetizing everything and does not allow code review).  Maybe it does work, maybe it doesnt.  Either way its not possible to determine if he actually uses pure JS.  I tend to believe it does NOT unless someone else has seen it's source and can confirm it does.  The source in the public repo is a WASM file,  its not pure JS.

**Comment by echel0n** [comment]
coinhive cryptonight looks rather customized so does xmrig's so not sure what would need changing as I don't have any background in c/c++ coding

**Comment by echel0n** [comment]
check this out https://github.com/notgiven688/webminerpool/tree/master/hash_cn/webassembly/xmr

**Comment by Lancelight** [comment]
Another cpp source to compare is:  https://github.com/fireice-uk/xmr-stak/blob/master/xmrstak/backend/cpu/crypto/cryptonight_common.cpp

**Comment by old-mikser** [comment]
@echel0n hoped you have. Because I don't have any experience with C/C++ also.

**Comment by Lancelight** [comment]
I have c/c++ exp but have never worked with WASM.  I assume its nothing more than some sort of SDK that compiles into a binary file that JS can access.  This may turn out to be something I'll need to do which I was hoping to avoid lol.

**Comment by old-mikser** [comment]
@Lancelight than if you have C bacground, can you using any of CLI miners listed by you above hash this blob - "0707b0cd9dd605d16dd9b0f5e59ce246ab39fd9413a889419b4a32385190c296e98f53f4a083470000000048eba682b41bcdd20572d5a43a585ced0ad4dd1723d0da5a79af17aa8f949eae0f"?

**Comment by Lancelight** [comment]
This webminerpool repo seems interesting,  it seems to already be setup to create a WASM file according to it's Makefile.  I could easily compile it and create a WASM file.  It has this in the Makefile " -s WASM=1"

**Comment by Lancelight** [comment]
Well I just setup EMCC which it needs to compile.  I'll see if I cant get it to spit out a wasm file.

**Comment by Lancelight** [comment]
Ok well I think I've managed to get it.  I also converted it to wat,  so you should be able to reconvert to wasm as well.  Here are both files just in case.  

http://crystaldawn.net/cn.wasm
http://crystaldawn.net/cn.wat

**Comment by echel0n** [comment]
I've decoded and attached the files cpufan.club uses, their site claims its implemented the new Monero v7 PoW and I've managed to get the code running but still no valid hashes are being generated from what I can see.

[cfc_decoded.zip](https://github.com/cazala/coin-hive-stratum/files/1888646/cfc_decoded.zip)

**Comment by echel0n** [comment]
i'm starting to think we need to make changes to the NodeJS app here on how it handles the hash after cryptonight is done with it before its sent back into the pool, I suspect it may need to be encoded to properly be accepted. Pulling apart the VB code for webminer might prove to show how thats done as well.

**Comment by focaeppe** [comment]
This guy find a solution but unfortunately he won't share for now
*** **V7 Upgrade** ***
As you know all the major cryptonight coins will switch (or have switched like Monero) to cryptonight V7 (or to other cryptonight implementations). In the meantime CoinHive have switched to a StO (Security through Obscurity) protected version of their webassembly/asmjs sources so they are not longer usable to mine in custom pools.
In order to make the miner resistant against this type of network upgrades I've almost rewritten the miner from scratch and compiled the webassembly/asmjs directly from the monero implementation. The miner now works fine with cryptonightv7 and will work fine with the future network upgrades.

https://batmine.club

**Comment by cazala** [comment]
Hi guys,

I couldn't spend time on this yet, but I spoke to Snipa about this and he told that `"Low difficulty share"` is basically an invalid share, it doesn't strictly have to do with the difficulty, it's just a wrong hash. So I think that kinda confirms what @slayerulan said (https://github.com/cazala/coin-hive-stratum/issues/167#issuecomment-379565539) and coinhive is obfuscating the blob before sending it over the wire. 

I'm on batmine's telegram and it looks like they managed to get it working, but they are not opensourcing it so ¯\\\_(ツ)_/¯ for the moment I'm busy with decentraland and I can't spend much time on this, maybe next weekend.

**Comment by slayerulan** [comment]
@cazala i reverse engineered batmine's aes+xor'ed code and they basically using webminerpool's code but claiming that they made it. you can use it freely: https://github.com/notgiven688/webminerpool

but the problem with that is it sends hashes too slow compared to coinhive :)

**Comment by Lancelight** [comment]
Hmm has anyone tried the wasm file that I compiled from the webminer repo?  I havent looked into its source much at all but I would assume it does not have the obfuscation that the coinhive version has.  All i did to test the wasm itself was used webminers own test page to see if it works on custom pools and it indeed does.  It sounds like this batmine thing may also have done the same thing we tried last night which was to compile the wasm directly from a monero cli miners source ( a version of which which can be found in webminers repo and is easily compiled with gcc+emcc).

**Comment by cazala** [comment]
@slayerulan I was just cloning that repo (:

**Comment by grmphf** [comment]
@LanceLight Yes i try it, but there are some memory problems to load this wasm. I investigated on that, but even i give lot of memory, the problem remains as before.

**Comment by PiTi2k5** [comment]
Feel free to test it.
Based on webminerpool source with a lot customizing ;)
https://www.monero.crypto-webminer.com/monero.html

**Comment by patrickbau** [comment]
@cazala 
Here is the source used in Batmine
https://github.com/notgiven688/webminerpool

**Comment by grmphf** [comment]
@PiTi2k5 That seems to be working. but it's 2-3 times slower than my test on coinhive

**Comment by PiTi2k5** [comment]
@scouinou you need a long term test. I run it for about 2 hrs and the pool stats almost equal as coinhive. The new algo is around 5% slower confirmed from official monero.
For old algo you can use the new coinhive script with customizing. It runs very good except for Graft pools. The blob sends 070 for new algo and you need to set version < 7 so old algo is used (or force the old algo)

**Comment by patrickbau** [comment]
@scouinou 
I just tested on 200 miners and i confirm that the hash is more then coinhive,

**Comment by patrickbau** [comment]
@Lancelight 
Yes i tested it it doesn't work :(

**Comment by Lancelight** [comment]
@patrickbau Hmm ok,  well i'll just keep working on it when I have more time.  The source for the wasm is going to be critical to have in an opensource webmining project if it expects to be successful.  Without it, forks are going to be major issues.

**Comment by slayerulan** [comment]
@Lancelight webminerpool already open sourced wasm?

**Comment by echel0n** [comment]
I can confirm webminerpool is up and running just fine.

**Comment by Lancelight** [comment]
He did yes,  thats where I swiped the source of cryptonight to compile the WASM file which some reported didnt seem to work.  But I did test his project and it does indeed pass shares to all of the pools I tried.  I'm not sure why the wasm didnt work,  maybe I copy/pasted something wrong but it does work with his project.  Maybe he uses a version of cryptonight from a completely different miner than what coinhive uses.  I have no idea where coinhive derives their wasm files from (and this project uses coinhive wasm files,  so the js files would be based on that assumption as well)

All I did was compile the cryptonight in his project with emcc which then created a file called "cn.html".  Inside that cn.html file is a base64 encoded version of the wasm file which I then copied,  de-encoded,  and then tested it with wasm2wat.  The wat file looked valid to me,  so I had posted it here (which I have since removed since people reported it wasnt working).

**Comment by slayerulan** [comment]
@Lancelight dude just shut up. please shut up you talking just nonsense on snipa22's issue and this issue. i wanted to keep calm but somebody needs to tell you to just stop. only think about it to tell that how retarded are u talking; why are you compiling an opensource project then decompiling (de encoding in your words) to reverse engineering it with wasm2wat? you can already review source code directly.

everyone; you can freely fork webminerpool source code; it works! they include wasm, frontend and backend on github. but backend is written in c# but you can easily edit the frontend to solve the hashes according to xmr-node-proxy+coin-hive-stratum. but the only problem is that webminerpool's code solving hashes 2x slower than coinhive. if you're ok with that then np at all.

if someone figures out how coinhive obfuscates hashes when sent to the server, please share it with us. crypto-loot and cpufan.club is already figured it out.

**Comment by Lancelight** [comment]
What the fuck are you babbling about.  I have absolutely never posted in any snipa22's projects.  Sounds to me like you have mental problems,  beginning with delusional fantasy posts that simply dont exist.  I dont see you compiling, modifying, and posting files/code snippets anywhere.  Put up or shut up.  The only thing I've seen you do is point to web resources.  Anyone with a monitor can do that.

**Comment by slayerulan** [comment]
@Lancelight ok

**Comment by ecSpl01t** [comment]
Someone resolve this problem ?

**Comment by Lancelight** [comment]
For this particular project,  not not yet.  The problem and fix is known,  now people are working on implementing the fix.  It may take awhile.

**Comment by cron13** [comment]
Hey, any solutions other than https://github.com/notgiven688/webminerpool ?

**Comment by galatolofederico** [comment]
Hey batmine admin here!
I'm using the same protocol as webminerpool for websocket comunication becouse I was inspired by their work and readjusted it (a lot!) to fit in my system.
@slayerulan off course I've not just copy&pasted it and you can understand it just looking at my service that offer a whole range of features that all the others webmining service does not have.
For example batmine works with cryptonight-heavy and as far as i know nobody except me support this algorithm so you can understand that if i'm capable of porting cryptonight-heavy on webasm doing the same thing for a well documented algorithm as cryptonight v7 is an easy task. 
If i could submit a PR to this repo to make it work with cn-v7 i would have done it, but the webasm and the client-side js of this repo is just a clever hacked version of coinhive and nobody except for coinhive's engineers have the C++ sources.
My system is hard coded for batmine and I cannot just put it on github as it is (unless I write a huge documentation on how to configure it and deploy it). In the future when I will close the service I'll rewrite it and create a easy to install and use version to put here

**Comment by Lancelight** [comment]
Erm. Just what exactly did you "port" to webasm?  Webasm is just a compiler,  it's not really a "language" persay.  You build something in c/c++ and then you simply compile it with the webasm output.  Done.  So what exactly would you need to port?

Here is the wiki on what "webasm" actually is.  So as you can see,  there is,  shall I say,  a bit of misinformation in your reply :P  

https://en.wikipedia.org/wiki/WebAssembly

Pay particularly close attention to this statement:  " Programs can be compiled from high-level languages into wasm modules and loaded as libraries from within JavaScript applets."

Now that we actually KNOW what webasm is,  here is the compiler used to compile c/c++ apps into webasm by using emcc:  https://kripken.github.io/emscripten-site/docs/tools_reference/emcc.html

So,  what exactly did you 'port' to webasm now that we know there is no "porting" necessary? :D

Compiling cryptonight's c/c++ source into wasm is as simple as 1 command line call to emcc.  Done.  Not exactly difficult :)

**Comment by galatolofederico** [comment]
Exactly it is not difficult at all, you just need to know how to compile and link stuff properly. With porting I mean that you don't only need the cn hash function but you need to make some adaptors to make it work with js (and in my case I also need a common interface for all the hashing function that I support). I'm sorry if I've not used the proper terms but I just want to be understandable by everyone in this thread.

**Comment by gtc51** [comment]
@galatolofederico 
Don't bother replying to @Lancelight.
He is just a local buffoon whom you can safely ignore.

**Comment by hitmanex** [comment]
@Lancelight what @galatolofederico is trying to say is you are a dumbass.

**Comment by Lancelight** [comment]
@hitmanex You should take lessons on how to read thread histories and you'd see that everything galatolofederico mentioned was already covered in my post about compiling the WASM.  You would also have learned that he wasnt replying to me at all and was trying to cover up his claim that he had created something, when in fact he had not.  He simply used Webminers project as pointed out by slayerulan.  Then I called him out on the attempted "O I ported something" bullshit, which is just that,  utter fucking bullshit.  He didnt port jack shit.  And his claim that "Hes the only project that supports CN Heavy" or whatever is nothing spectacular.  Just compile CN Heavy using 1 command line command with EMCC and there you have it,  a WASM that supports CN Heavy.  Wow.  Thats a lotta work.  Write up a quickie 10 or 15 line JS file that uses the new functions (if there are even that many that need to be used) and you can slap a price tag on it,  then reply in threads defending yourself for swiping others work and claiming it as your own (ala galatolofederico) like you have made some amazing new feature.

Anyone who does not understand this fact (such as GTC51 and hitmanex) would fail to see what I am even talking about.  If you look up above,  I've already made notes that this project uses CH's compiled WASM and that in order for this to ever work again, they either need the sources from CH's new WASM (which isnt really possible since it's now obfuscated beyond practical use) OR compile it themselves (which I described in detail how to do and successfully did so myself as well right here in this thread).  I then posted the results of that and no one understood what to do with it, so it went no where and I simply deleted the WASM file since its clear no one is going to know how to use it and I'm not about to do support for how to use it.  I then found a new project suggested here in this thread by echel0n who is the only other person in this thread who even has a clue on how to fix things like this other than myself.  I didnt see any point in re-inventing the wheel,  so I no longer use this garbage project.  I just use webminer's project as a fork instead (which is the same conclusion that Galatolofederico came to).

Therefore,  you missed the fact that Galatolofederico was simply pointing at a project that was already identified by echel0n and myself as being a better project to use since it had working CN C/C++ sources from which to compile a WASM from and it also had a nice JS file already to go that uses the functions that the WASM provides (which in turn came from the c/c++ sources of a cli miner,  webminer simply dropped them in his repo so that its there for others to compile the wasm themselves).  Therefore,  this project is now 100% dead as far as I'm concerned.  It's obsoleted by weminers project.  His post didnt bring in anything new that had already been covered by me and echel0n.  However it did appear like he was shilling his own service (which he probably charges for and could be why he posted here to begin with).  I dont take kindly to shills who shill their own pay services.  I've never even looked at "Batmine" or whatever it is but I wouldnt be one bit surprised if there was some sort of pay model somewhere within it for something.

**Comment by hitmanex** [comment]
@Lancelight you're wrong. also didn't read

**Comment by Lancelight** [comment]
LMFAO.  Well this should be amusing.  Prove it.

**Comment by hitmanex** [comment]
you prove it

**Comment by Lancelight** [comment]
Lol ya thatz what I thought.  Go crawl back under the rock where you belong.

Exerptz from above.  Your turn........  *crickets*.  Geee like that wasnt predictable at all.  LMFAO.

@ cazala **i reverse engineered batmine's aes+xor'ed code and they basically using webminerpool's code but claiming that they made it**. you can use it freely: https://github.com/notgiven688/webminerpoolbut the problem with that is it sends hashes too slow compared to coinhive :)
--

 **This includes coinhive and all of it's derivatives. This will be the case until coinhive or one of its derivatives releases a fix for the new fork for XMR's new hashing algo**

**Comment by hitmanex** [comment]
@Lancelight everything you said is wrong. LMFAO!!!!

**Comment by Lancelight** [comment]
Ahh I see, so whenever you've been proven a moron you just revert to broken rec.... err retardism.  Got it.

**Comment by neverknoww** [comment]
how about stop picking on each other and try to come up with a solution.

**Comment by hitmanex** [comment]
you expect dumb boy lancelight to have a solution?

i on the other hand have been mining this whole time

**Comment by brainscar** [comment]
What can we do to fix this?

**Comment by hitmanex** [comment]
be smart like @Lancelight .. oh wait he never fixed it to begin with

**Comment by slayerulan** [comment]
you guys, i already said the solution too many times. JUST USE F#CKING notgiven688/webminerpool 

you can easily integrate it to snipa22's proxy btw

**Comment by marcelaodev** [comment]
@slayerulan how to integrate to snipa's ?

**Comment by cron13** [comment]
notgiven688/webminerpool two times slower than this solution. :-(
Maybe it possible to fix "low diff share" problem?

**Comment by hitmanex** [comment]
@Lancelight is the braniac here we must ask him..unless all hes good for is writing paragraphs of useless crap

**Comment by EZForever** [comment]
Hello everyone,
I've found the source of this issue: the CryptoNight library implemented by CoinHive returns different (maybe encrypted) result than other libraries.
So maybe we could replace the one script is using with a known clean version? That should fix the problem.
See [here](https://github.com/EZForever/libcryptohive) for more details.

btw, arguing is bad :(

**Comment by ghost** [comment]
There is no need for us to argue, Although cazala may don't have enough time to fix this problem, I believe we can solve it, this is not a good place to chat, Let's use telegram [t.me/coinhivestratum](t.me/coinhivestratum) instead.
